src/pkg/cluster/tunnel.go

```diff
@@ -352,38 +351,16 @@ func (tunnel *Tunnel) Wrap(function func() error) error {
 
 // Connect will establish a tunnel to the specified target.
 func (tunnel *Tunnel) Connect(ctx context.Context) (string, error) {
-	url, err := tunnel.establish(ctx)
-
-	// Try to establish the tunnel up to 3 times.
-	if err != nil {
-		tunnel.attempt++
-
-		// If we have exceeded the number of attempts, exit with an error.
-		if tunnel.attempt > 3 {
-			return "", fmt.Errorf("unable to establish tunnel after 3 attempts: %w", err)
-		}
-
-		// Otherwise, retry the connection but delay increasing intervals between attempts.
-		delay := tunnel.attempt * 10
-		message.Debugf("%s", err.Error())
-		message.Debugf("Delay creating tunnel, waiting %d seconds...", delay)
-
-		timer := time.NewTimer(0)
-		defer timer.Stop()
-
-		select {
-		case <-ctx.Done():
-			return "", ctx.Err()
-		case <-timer.C:
-			url, err = tunnel.Connect(ctx)
-			if err != nil {
-				return "", err
-			}
-
-			timer.Reset(time.Duration(delay) * time.Second)
+	url, err := retry.DoWithData(func() (string, error) {
+		url, err := tunnel.establish(ctx)
+		if err != nil {
+			return "", err
 		}
+		return url, nil
+	}, retry.Context(ctx), retry.Attempts(3))
```

Comment:
Not sure if it's important, but the amount of delay here is changing significantly. There was about 30 seconds total delay before, now there is about 300 milliseconds total delay.

Reply:
Will increase the amount of attempts instead to fix this.